From 6526777baf6a3b51a3569836f7cbe0b14dc39aea Mon Sep 17 00:00:00 2001
From: Ulrich Weigand <ulrich.weigand@de.ibm.com>
Date: Wed, 10 Jul 2024 23:21:57 +0200
Subject: [PATCH] core: Limit remaining f16 doctests to x86_64 linux

On s390x, every use of the f16 data type will currently ICE
due to https://github.com/llvm/llvm-project/issues/50374,
causing doctest failures on the platform.

Most doctests were already restricted to certain platforms,
so fix this by likewise restricting the remaining five.
---
 library/core/src/num/f16.rs | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/library/core/src/num/f16.rs b/library/core/src/num/f16.rs
index c4d4584544b..b44ea3bf2ef 100644
--- a/library/core/src/num/f16.rs
+++ b/library/core/src/num/f16.rs
@@ -227,12 +227,15 @@ pub const fn is_nan(self) -> bool {
     ///
     /// ```
     /// #![feature(f16)]
+    /// # // FIXME(f16_f128): LLVM crashes on s390x, llvm/llvm-project#50374
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let f = 7.0_f16;
     /// let g = -7.0_f16;
     ///
     /// assert!(f.is_sign_positive());
     /// assert!(!g.is_sign_positive());
+    /// # }
     /// ```
     #[inline]
     #[must_use]
@@ -250,12 +253,15 @@ pub fn is_sign_positive(self) -> bool {
     ///
     /// ```
     /// #![feature(f16)]
+    /// # // FIXME(f16_f128): LLVM crashes on s390x, llvm/llvm-project#50374
+    /// # #[cfg(all(target_arch = "x86_64", target_os = "linux"))] {
     ///
     /// let f = 7.0_f16;
     /// let g = -7.0_f16;
     ///
     /// assert!(!f.is_sign_negative());
     /// assert!(g.is_sign_negative());
+    /// # }
     /// ```
     #[inline]
     #[must_use]
-- 
2.46.0

